### PR TITLE
initial open source commit

### PR DESCRIPTION
**Pull request by hongchaodeng**
This is the initial version that we have inside Alibaba and have a couple of users used already.

### CODE REVIEW

**Comment by hongchaodeng** [comment]
Also credited to @Diddaa and @wonderflow for initial hard work!

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module gitlab.alibaba-inc.com/cloudnativeapp/oam-runtime
```
Comment (nerdyyatrice) [inline go.mod]: is this supposed to be an alibaba module?

Comment (hongchaodeng) [reply go.mod]: Good catch. Fixing it shortly.

**Comment by hongchaodeng** [comment]
I have already discussed this with @wonderflow @resouer, received general agreement from our initial users.

Merging this to keep the work forward. We will be the initial responsible persons to own, develop, and maintain this code.

**Pull request merged**